xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -318,6 +328,39 @@ void cancelEndpointDataWatch(String clusterName, EndpointWatcher watcher) {
     }
   }
 
+  @Override
+  LoadReportClient reportClientStats(String clusterName, String serverUri) {
+    checkArgument(serverUri.equals(""),
```

Comment:
nit:  add `checkNotNull(serverUri, "null-serverUri indicates disabling stats report")`.

Reply:
I don't think that's necessary to have both `checkArgument` and `checkNotNull` here. It will break `checkArgument` immediately if `serverUri` is `null`, and we can get `NullPointException` error to see what goes wrong.

Reply:
Then that NPE is without a string description but only a line number and stack trace, which is hard to debug if you don't know the version of the source code.

An alternative is to use `"".equals(serverUri)` to avoid NPE.

Reply:
Alright, I added `checkNotNull` for `serverUri`. 

Btw, your point is to explicitly populate most accurate error messages. `"".equals(serverUri)` makes `checkArgument` fail if `serverUri` is `null`. But still, you don't know from the stacktrace that it was because `serverUri` is `null`. So added `checkNotNull`.